developer-docs-site/docs/move/move-on-aptos/cryptography.md

```diff
@@ -0,0 +1,186 @@
+2. When native functions are not available, **empower developers** to implement their own cryptographic primitives on top of abstract cryptographic building blocks such as _finite fields_ and _Abelian groups_ (e.g., via the [`aptos_std::crypto_algebra`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/crypto_algebra.move) module).
+
+An overview of these features and (some of) the applications they enable follows below. 
+For a detailed view, see the [cryptography Move modules code](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/aptos-stdlib/sources/cryptography).
+
+## Cryptographic hash functions in Move
+
+Developers can now use more cryptographic hash functions in Move via the [`aptos_std::aptos_hash`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/hash.move) module:
+
+ 1. Keccak256
+ 2. SHA2-512
+ 3. SHA3-512
+ 4. RIPEMD160
+ 5. Blake2b-256
+
+Some of these functions can be used for interoperability with other chains (e.g., verifying Ethereum Merkle proofs via [`aptos_std::aptos_hash::keccak256`](https://github.com/aptos-labs/aptos-core/blob/137acee4c6dddb1c86398dce25b041d78a3028d3/aptos-move/framework/aptos-stdlib/sources/hash.move#L35)).
+Others, have lower gas costs, such as [`aptos_std::aptos_hash::blake2b_256`](https://github.com/aptos-labs/aptos-core/blob/137acee4c6dddb1c86398dce25b041d78a3028d3/aptos-move/framework/aptos-stdlib/sources/hash.move#L69).
+In general, a wider variety of hash functions give developers additional freedom in terms of both security and interoperability with other off-chain cryptographic system.
+
+## Digital signature verification in Move
+
+Developers can now use a *type-safe* API for verifying many types of digital signatures in Move:
+
+ 1. **Ed25519** signatures in [`aptos_std::ed25519`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move)
+ 2. “Naive” $t$-out-of-$n$ **Ed25519** **multi**-signatures in [`aptos_std::multi_ed25519`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.move) (via simple concatentation of Ed25519 signatures)
+ 3. **ECDSA** signatures over secp256k1 in [`aptos_std::secp256k1`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/secp256k1.move)
+ 4. **BLS** signatures, **multi**-signatures, **aggregate** signatures and **threshold** signatures over BLS12-381 in [`aptos_std::bls12831`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move)
```

Comment:
maybe we could make a table here on the pros and cons of these different pieces?

I'd love to push users away from using a key algorithm out of convenience and more from whatever education we can provide them.

Reply:
Hm... that would be a rather nuanced analysis. There's a lot of dimensions to analyze for "pros" and "cons."

I can try to think of some relevant things.

Reply:
Fixed.